applets/clock/clock.c

```diff
@@ -901,8 +906,74 @@ position_calendar_popup (ClockData *cd)
         int             i, n;
         gboolean        found_monitor = FALSE;
 
-        if (!GDK_IS_X11_DISPLAY (gdk_display_get_default ()))
-                return;
+
+        if (GDK_IS_WAYLAND_DISPLAY (gdk_display_get_default ()))
```

Comment:
and if the X11 and Wayland code don't share much logic, you could have helpers for each so the body of that one is just the selector calling the right one… just a thought, because it gets big.

Reply:
`#ifdef HAVE_X11` compile time selectors have been added to the x11 portion of the code as well as a runtime 
`if (GDK_IS_X11_DISPLAY (gdk_display_get_default ()))` selector. Therefore, if the panel is built with only x11 or with only wayland support (allowed since the initial wayland commits a while ago), only the relevent code is built. In builds supporting both, only the x11 or wayland code is selected at runtime, so this has been implemented.

Thanks for your work and your help with the wayland support, I appreciate it